File: packages/gbnf-experiment/data/20260909T151143Z_170b737d/ported_implementation/src/utils/errors/get_input_as_string.ts
import type { ValidInput } from "./errors_types.ts";

export const get_input_as_string = (src: ValidInput): string => {
  if (typeof src === "string") {
    return src;
  }
  if (typeof src === "number") {
    return String.fromCodePoint(src);
  }
  return src.map((cp) => String.fromCodePoint(cp)).join("");
};

File: packages/gbnf-experiment/data/20260909T015142Z_f8ca89ff/ported_implementation/utils/errors/get_input_as_string.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:  # pragma: no cover
    from ...grammar_graph.types import ValidInput

__all__ = ["get_input_as_string", "getInputAsString"]


def get_input_as_string(src: "ValidInput") -> str:
    if isinstance(src, str):
        return src
    code_points = list(src) if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(cp) for cp in code_points)


getInputAsString = get_input_as_string
